ui/apps/platform/src/Containers/Collections/CollectionResults.tsx

```diff
@@ -1,7 +1,157 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState, useMemo } from 'react';
+import {
+    Button,
+    Divider,
+    EmptyState,
+    EmptyStateIcon,
+    EmptyStateVariant,
+    Flex,
+    FlexItem,
+    Select,
+    SearchInput,
+    SelectOption,
+    debounce,
+} from '@patternfly/react-core';
+import { ListIcon } from '@patternfly/react-icons';
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+import ResourceIcon from 'Components/PatternFly/ResourceIcon';
 
-function CollectionResults() {
-    return <></>;
+import { dryRunCollection, CollectionRequest } from 'services/CollectionsService';
+import { ListDeployment } from 'types/deployment.proto';
+import { SelectorEntityType } from './types';
+
+export type CollectionResultsProps = {
+    dryRunConfig: CollectionRequest;
+};
+
+function CollectionResults({ dryRunConfig }: CollectionResultsProps) {
+    const { isOpen, onToggle, closeSelect } = useSelectToggle();
+    const [selected, setSelected] = useState<SelectorEntityType>('Deployment');
+    const [filterText, setFilterText] = useState<string>('');
+    const [isEndOfResults, setIsEndOfResults] = useState<boolean>(false);
+    const [deployments, setDeployments] = useState<ListDeployment[]>([]);
+
+    const currentPage: number = Math.floor((deployments.length - 1) / 10);
+    const selectorRulesExist =
+        dryRunConfig.resourceSelectors?.[0]?.rules?.length > 0 ||
+        dryRunConfig.embeddedCollectionIds.length > 0;
+
+    function onRuleOptionSelect(_, value): void {
+        setSelected(value);
+        setFilterText('');
+        closeSelect();
+    }
+
+    const fetchDryRun = useCallback(
+        (page: number) => {
+            const pageSize = 10;
+            const query = { [selected]: filterText };
+            const { request } = dryRunCollection(dryRunConfig, query, page, pageSize);
+            request
+                .then((results) => {
+                    setIsEndOfResults(results.length < 10);
+                    setDeployments((current) => (page === 0 ? results : [...current, ...results]));
+                })
+                .catch(() => {
+                    // TODO: indicate results not loading properly?
+                });
+        },
+        [dryRunConfig, filterText, selected]
+    );
+
+    useEffect(() => {
+        if (selectorRulesExist) {
+            fetchDryRun(0);
+        }
+    }, [dryRunConfig, fetchDryRun, selectorRulesExist]);
+
+    const onSearchInputChange = useMemo(
+        () =>
+            debounce((value: string) => {
+                setFilterText(value);
+            }, 800),
+        []
+    );
+
+    return (
+        <>
+            <Divider />
+            {!selectorRulesExist ? (
+                <EmptyState variant={EmptyStateVariant.xs}>
+                    <EmptyStateIcon icon={ListIcon} />
+                    <p>
+                        Add selector rules or attach existing collections to view resource matches
```

Comment:
For my understanding, no rules implies create/clone/edit and not view, correct?

Reply:
No rules just implies not having made any changes to the "Collection Rules" section – I probably should've included that section in the screenshots

Reply:
I updated the screenshots to give a better representation of the logic